frontend/src/modules/widget/widget-utility.js

```diff
@@ -0,0 +1,93 @@
+import {
+  ALL_TIME_PERIOD_FILTER,
+  WIDGET_PERIOD_OPTIONS,
+  WIDGET_GRANULARITY_OPTIONS,
+  WEEKLY_GRANULARITY_FILTER,
+  SEVEN_DAYS_PERIOD_FILTER,
+  MONTHLY_GRANULARITY_FILTER,
+  FOURTEEN_DAYS_PERIOD_FILTER,
+  THREE_MONTHS_PERIOD_FILTER,
+} from './widget-constants';
+
+export const getSelectedPeriodFromLabel = (
```

Comment:
Could you add a small description on top of this methods to clarify what they do?
For this first method, why do we need:
```
if (granularity) {
    selectedPeriod = getGranularityBasedPeriod(granularity, selectedPeriod);
  }

  if (selectedPeriod && comparePeriods(selectedPeriod, defaultMinimumPeriod)) {
    return selectedPeriod;
  }
```

Couldn't we just do a find by label and if it doesn't exist, return the default value?

Reply:
I have added the comments to the function, also just to explain here, the granularity check is being done so that we can avoid cases where the user has changed the url with a mismatching selected period and granularity for eg providing the period as 7 days and the granularity as monthly will result in granularity being selected as monthly with no selected period value on the UI

Reply:
Both these checks were made in order to tackle the user manipulation in the url